### PR TITLE
solana-ibc: add update_stake call altering stake of multiple validators

### PR DESCRIPTION
**Pull request by dhruvja**
Introduce update_stake method which allows changing stake of multiple
validators in a single go.  It can be called with CPI from the
restaking program only.

### CODE REVIEW

File: solana/solana-ibc/programs/solana-ibc/src/chain.rs [hunk 1]
```diff
@@ -173,6 +174,42 @@ impl ChainData {
...
     }
 
+    /// Updates multiple validators stake based on the operation
+    ///
+    /// Fails if candidate is not found during remove operation
+    /// or if the subtraction overflows
+    pub fn set_stake_multiple(
+        &mut self,
+        validator_change_in_stake: Vec<(Pubkey, u128)>,
+        operation: StakeOperation,
```
Comment (mina86) [inline solana/solana-ibc/programs/solana-ibc/src/chain.rs:184]: I think we can get away with having i128 in the map and then we don’t need StakeOperation.

**Review by mina86 — COMMENTED** [review]
Also, this PR can be more efficient with https://github.com/ComposableFi/emulated-light-client/pull/337 I can update it once #337 gets merged.

**Comment by mina86** [comment]
I’ve pushed the changes. This is blocked on #337 but once that’s merged this one can go I think. It is somewhat simpler.

**Comment by dhruvja** [comment]
I have merged #337

**Review by mina86 — APPROVED** [review]

**Pull request merged**